### DISCUSSION TITLE
Question Regarding GGML_MAX_CONTEXTS

### DISCUSSION THREAD

**Discussion by bradmit** [ggml-org/whisper.cpp#2520] [Q&A]
Hey there.  In previous versions of the lib, I've raised this from 64 up to 256 as I actually ran into a situation with the way I'm using the lib that it was crashing (poor coding on my part) due to running out of contexts.
Is there any issue with raising this from 64 -> 256?

This is the CPU implementation - ggml.h - GGML_MAX_CONTEXTS

Thanks

**Comment by ggerganov** [comment]
It's not a problem to increase `GGML_MAX_CONTEXTS` - it will just increase a bit the static global memory usage, but nothing concerning overall.

**Reply by KitaitiMakoto** [reply]
It's reasonable there's some limitation to the number of contexts. But, whisper.cpp should handle `NULL` appropriately if `ggml_init()` returns `NULL`. Now it causes segmentation fault because it handle `NULL` as normal `ggml_context`.

For instance, is it okay to return `NULL` from functions which call `ggml_init()` when it returns `NULL`?
